vhdbuilder/packer/cleanup.sh

```diff
@@ -193,4 +245,22 @@ if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${DRY_RUN}" == "False" ]]; then
           echo "Deletion completed"
       fi
   done
-fi
\ No newline at end of file
+fi
+
+# clean up old packer and vnet resource groups created over a week ago that didn't get cleaned in the building vhd step in case the pipeline fails
+if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${DRY_RUN}" == "True" ]]; then
```

Comment:
I'd consider cleaning up old pkr resource groups irregardless of dry-run

Reply:
hmm, I'm actually wondering if it would make more sense to have a separate pipeline variable that we could use to skip the image/SIG/image-definition deletion backfill steps, as opposed to overloading the functionality/meaning of dry-run mode. I feel like dry-run mode should just determine the nature of the output of the build itself, rather than the other miscellaneous cleanup it does during this step.

Basically, the new pipeline variable would default to false on the production pipelines, and default to true on the testing pipelines (which would also be able to be overridden). If the variable is set to true, we don't do the backfill deletion, otherwise we always attempt the backfill deletion. The variable could be named something like SKIP_RESOURCE_DELETION_BACKFILL or something.

this would be something we can maybe swing back around to after this PR is merged in